File: leetcode/easy/496-next-greater-element-1.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        index_checker = {x: idx for idx, x in enumerate(nums2)}
        result = []
        for elem in nums1:
            current_index = index_checker.get(elem) + 1
            while(current_index <= len(nums2)):
                if current_index >= len(nums2):
                    result.append(-1)
                    break
                if nums2[current_index] > elem:
                    result.append(nums2[current_index])
                    break
                current_index += 1
        return result

File: leetcode/easy/496-next-greater-element-1.js
var nextGreaterElement = function(nums1, nums2) {
    const indexChecker = {}
    const result = [];
    for(let i = 0; i < nums2.length; i++) {
        indexChecker[nums2[i]] = i
    }
    for(elem of nums1) {
        let currentIndex = indexChecker[elem] + 1;
        while(currentIndex <= nums2.length) {
            if(currentIndex >= nums2.length) {
                result.push(-1);
                break;
            }
            if(nums2[currentIndex] > elem) {
                result.push(nums2[currentIndex]);
                break;
            }
            currentIndex++;
        }
    }
    return result;
};
